lib/Dialect/XTenNN/IR/XTenNNOps.cpp

```diff
@@ -552,11 +552,21 @@ LogicalResult TopK::inferReturnTypeComponents(
 
   auto inTy = cast<RankedTensorType>(adaptor.getInput().getType());
 
-  auto axis = adaptor.getAxis();
-  if (axis >= (uint64_t)inTy.getRank()) {
-    return emitOptionalError(location, "expected axis <= rank of input");
+  auto axis = (int64_t)adaptor.getAxis();
+  // onnx spec: axis: [-r, r-1]
+  if (!(axis >= -inTy.getRank()) || !(axis < inTy.getRank())) {
```

Comment:
Nit for readability:
```suggestion
  if (axis < -inTy.getRank() || axis => inTy.getRank())) {
```